### PR TITLE
Added new practitioner identifiers to FHIR payload. Provider National ID and Licence number

### PR DESCRIPTION
**Pull request by PatrickWaweru**
Added new practitioner identifiers to FHIR payload. Provider National ID and Licence number

### CODE REVIEW

**Review by makombe — APPROVED** [review]
Thanks @PatrickWaweru

File: omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java [hunk 1]
```diff
@@ -92,6 +93,12 @@ public InsuranceClaim createClaim(NewClaimForm form) {
 
         System.out.println("Insurance Module: Encounter UUID : " + form.getEncounterUuid());
         Encounter encounter = Context.getEncounterService().getEncounterByUuid(form.getEncounterUuid());
+        // Check for null encounter and if null get the first patient encounter TODO: remove this
+        if(encounter == null) {
+            System.out.println("Insurance Module: Encounter : Fail control. Encounter was null so we use a random one");
```
Comment (makombe) [inline omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java:98]: Maybe we need to remove some of the print logs. They can be noisy at times

**Review by patryllus — APPROVED** [review]
Looks okay

File: omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java (hunk 1 above)
Comment (patryllus) [reply omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java:98]: @makombe We need the logs at the moment for testing integrations.\

File: omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java [hunk 2]
```diff
@@ -102,7 +109,15 @@ public InsuranceClaim createClaim(NewClaimForm form) {
         nextClaim.setStatus(InsuranceClaimStatus.ENTERED);
         nextClaim.setLocation(getClaimLocation(form));
 
+        System.out.println("Insurance Module: Provider UUID : " + form.getProvider());
         Provider provider = Context.getProviderService().getProviderByUuid(form.getProvider());
+        // Check for null provider and if null get the first registered provider TODO: remove this
+        if(provider == null) {
+            System.out.println("Insurance Module: Provider : Fail control. Provider was null so we use a random one");
+            provider = Context.getProviderService().getAllProviders().get(0);
```
Comment (ojwanganto) [inline omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java:117]: I don't agree with this though. I hope the provider search widget can supply this if need be but defaulting to any provider is wrong

File: omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java [hunk 3]
```diff
@@ -92,6 +93,12 @@ public InsuranceClaim createClaim(NewClaimForm form) {
 
         System.out.println("Insurance Module: Encounter UUID : " + form.getEncounterUuid());
         Encounter encounter = Context.getEncounterService().getEncounterByUuid(form.getEncounterUuid());
+        // Check for null encounter and if null get the first patient encounter TODO: remove this
+        if(encounter == null) {
+            System.out.println("Insurance Module: Encounter : Fail control. Encounter was null so we use a random one");
+            List<Encounter> encounters = Context.getEncounterService().getEncountersByPatient(patient);
+            encounter = encounters.size() > 0 ? encounters.get(encounters.size() - 1) : null;
```
Comment (ojwanganto) [inline omod/src/main/java/org/openmrs/module/insuranceclaims/forms/impl/ClaimFormServiceImpl.java:100]: Even this is wrong. As a way of provenance, one must have had an encounter during the period of the bill i.e. the associated visit.

**Pull request merged**